tools/src/main/java/org/apache/kafka/tools/StreamsCleanupClient.java

```diff
@@ -0,0 +1,208 @@
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package org.apache.kafka.tools;
+
+import joptsimple.OptionException;
+import joptsimple.OptionParser;
+import joptsimple.OptionSet;
+import joptsimple.OptionSpec;
+import kafka.admin.TopicCommand;
+import kafka.utils.ZkUtils;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.security.JaasUtils;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+
+/**
+ * {@link StreamsCleanupClient} resets a Kafka Streams application to the very beginning. Thus, an application can
+ * reprocess the whole input from scratch.
+ * <p>
+ * This includes, setting source topic offsets to zero, skipping over all intermediate user topics,
+ * and deleting all internally created topics.
+ */
+public class StreamsCleanupClient {
+    private static OptionSpec<String> bootstrapServerOption;
+    private static OptionSpec<String> zookeeperOption;
+    private static OptionSpec<String> applicationIdOption;
+    private static OptionSpec<String> sourceTopicsOption;
+    private static OptionSpec<String> intermediateTopicsOption;
+
+    private OptionSet options = null;
+
+    public void run(String[] args) {
+        ZkUtils zkUtils = null;
+
+        int exitCode = 0;
+        try {
+            parseArguments(args);
+
+            zkUtils = ZkUtils.apply(options.valueOf(zookeeperOption),
+                    30000,
+                    30000,
+                    JaasUtils.isZkSecurityEnabled());
+
+            final List<String> allTopics = scala.collection.JavaConversions.seqAsJavaList(zkUtils.getAllTopics());
+
+            resetSourceTopicOffsets();
+            seekToEndIntermediateTopics(zkUtils, allTopics);
+            deleteInternalTopics(zkUtils, allTopics);
+        } catch (Exception e) {
+            exitCode = -1;
+            System.err.println(e.getMessage());
+        } finally {
+            if (zkUtils != null) {
+                zkUtils.close();
+            }
+            System.exit(exitCode);
+        }
+    }
+
+    private void parseArguments(final String[] args) throws IOException {
+        final OptionParser optionParser = new OptionParser();
+        applicationIdOption = optionParser.accepts("application-id", "The Kafka Streams application ID (application.id)")
+                .withRequiredArg()
+                .describedAs("id")
+                .ofType(String.class)
+                .required();
+        bootstrapServerOption = optionParser.accepts("bootstrap-server", "Format: <host:port>")
```

Comment:
I designed this to only accept a single server `host:port` parameter (its singular, not plural). Do you think this is a problem, because it does not align with other Kafka tools?